tests/src/providers/testqgswmsprovider.cpp

```diff
@@ -243,6 +247,52 @@ class TestQgsWmsProvider: public QObject
       QVERIFY( provider.layerMetadata().rights().at( 0 ).startsWith( "Base map and data from OpenStreetMap and OpenStreetMap Foundation" ) );
     }
 
+    void testConvertToValue()
+    {
+      QString dataDir( TEST_DATA_DIR );
+
+      QgsXyzConnection xyzConn;
+      xyzConn.url = QUrl::fromLocalFile( dataDir + QStringLiteral( "/maptiler_terrain_rgb_1.png" ) ).toString();
+      QgsRasterLayer layer( xyzConn.encodedUri(), "terrain", "wms" );
+      QVERIFY( layer.isValid() );
+      QVERIFY( layer.dataProvider()->dataType( 1 ) == Qgis::DataType::ARGB32 );
+
+      xyzConn.encodingSchemeKey = QStringLiteral( "maptilerterrain" );
+      QgsRasterLayer layer2( xyzConn.encodedUri(), "terrain", "wms" );
+      QVERIFY( layer2.isValid() );
+      QVERIFY( layer2.dataProvider()->dataType( 1 ) == Qgis::DataType::Float32 );
+
+      QgsMapSettings mapSettings;
+      mapSettings.setLayers( QList<QgsMapLayer *>() << &layer2 );
+      mapSettings.setExtent( layer2.extent() );
+      mapSettings.setOutputSize( QSize( 400, 400 ) );
+      mapSettings.setOutputDpi( 96 );
+      mapSettings.setDpiTarget( 48 );
+      QVERIFY( imageCheck( "convert_value_1", mapSettings ) );
+
+
+      xyzConn.url = QUrl::fromLocalFile( dataDir + QStringLiteral( "/maptiler_terrain_rgb_2.png" ) ).toString();
+      QgsRasterLayer layer3( xyzConn.encodedUri(), "terrain", "wms" );
+      QVERIFY( layer3.isValid() );
+      QVERIFY( layer3.dataProvider()->dataType( 1 ) == Qgis::DataType::Float32 );
+
+      // change renderer
+      std::unique_ptr<QgsColorRampShader> colorRampShader( new QgsColorRampShader( 0, 9000, QgsStyle::defaultStyle()->colorRamp( QStringLiteral( "Viridis" ) ) ) );
+      colorRampShader->classifyColorRamp( 10, -1 );
+
+      std::unique_ptr<QgsRasterShader> shader( new QgsRasterShader( 0, 9000 ) );
+      shader->setRasterShaderFunction( colorRampShader.release() );
+      shader->setRasterShaderFunction( colorRampShader.release() );
+      layer3.setRenderer( new QgsSingleBandPseudoColorRenderer( layer3.renderer()->input(), 1, shader.release() ) );
+
+      mapSettings.setLayers( QList<QgsMapLayer *>() << &layer3 );
+      mapSettings.setExtent( layer3.extent() );
+      mapSettings.setOutputSize( QSize( 400, 400 ) );
+      mapSettings.setOutputDpi( 96 );
+      mapSettings.setDpiTarget( 48 );
+      QVERIFY( imageCheck( "convert_value_2", mapSettings ) );
```

Comment:
They suggested to use only a single tile in unit tests instead of two, and they prefer not to use the root tile, but rather something at least a bit zoomed in. So how about e.g. https://api.maptiler.com/tiles/terrain-rgb/4/11/6.png - it includes high mountains in Himalayas and also a bit of ocean...